methods/inputs/filter_gedi_data.py

```diff
@@ -9,85 +9,125 @@
-        df = parse_file(os.path.join(gedi_directory_path, path))
+            # drop any columns not in the interesting_columns set
+            df.drop(list(set(df.columns) - interesting_columns), axis=1)
 
-        # Drop data we don't care about
-        # The date limits here come from discussion with Miranda Lam:
-        # "we currently use the JRC Annual Change 2020 for land cover class and GEDI shots from 2020/1/1 to 2021/1/1"
-        df.drop(list(set(df.columns) - interesting_columns), axis=1)
-        df.drop(df.index[df.beam_type != "full"], inplace=True)
-        df.drop(df.index[df.degrade_flag != 0], inplace=True)
-        df.drop(df.index[df.l4_quality_flag != 1], inplace=True)
-        df.drop(df.index[df.absolute_time < datetime(2020, 1, 1, tzinfo=timezone.utc)], inplace=True)
-        df.drop(df.index[df.absolute_time >= datetime(2021, 1, 1, tzinfo=timezone.utc)], inplace=True)
-        df.drop(df.index[~df.within(boundary.geometry.values[0])], inplace=True)
+            # filter rows based on beam type, quality flag, degradation flag, and date range
+            df.drop(df.index[df.beam_type != "full"], inplace=True)
+            df.drop(df.index[df.degrade_flag != 0], inplace=True)
+            df.drop(df.index[df.l4_quality_flag != 1], inplace=True)
 
-        granules.append(df)
+            # filter based on whether the data point falls within the boundary geometry
+            df.drop(df.index[~df.within(boundary.geometry.values[0])], inplace=True)
 
+            granules.append(df)  # add the filtered DataFrame to the list
+
+    # concatenate all filtered granules into a single GeoDataFrame
     superset = gpd.GeoDataFrame(pd.concat(granules, ignore_index=True))
+    # write the result to a GeoJSON file
     superset.to_file(output_filename, driver="GeoJSON")
 
+# main function to handle argument parsing and script execution
 def main() -> None:
-    parser = argparse.ArgumentParser(description="Takes a set of GEDI granules and filters down" \
-        " the data to just what is needed for the AGB calculation")
+    # set up argument parsing
+    parser = argparse.ArgumentParser(description="takes a set of gedi granules and filters down"
+                                                 " the data to just what is needed for the agb calculation")
+    # add argument for the directory containing GEDI granules
+    parser.add_argument(
+        "--granules",
+        type=str,
+        required=True,
+        dest="gedi_directory_path",
+        help="location of gedi granules"
+    )
+    # add argument for the buffer boundary file
     parser.add_argument(
         "--buffer",
         type=str,
         required=True,
         dest="buffer_boundary_filename",
-        help="Buffer boundary file"
+        help="buffer boundary file"
     )
+    # add argument for the text file containing the list of granules to process
     parser.add_argument(
-        "--granules",
+        "--granule-list",
```

Comment:
Don't we need to update scripts/tmfpython.sh to match this?

Reply:
Yes, I'll make the additions to the shellscript so its consistent

Reply:
Thanks. I know @epingchris uses that script a lot, so not updating it would impact his workflow.